server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTranslogOperationsRequest.java

```diff
@@ -31,16 +32,14 @@
 
 public class RecoveryTranslogOperationsRequest extends TransportRequest {
 
-    private long recoveryId;
-    private ShardId shardId;
-    private List<Translog.Operation> operations;
-    private int totalTranslogOps = RecoveryState.Translog.UNKNOWN;
-    private long maxSeenAutoIdTimestampOnPrimary;
-    private long maxSeqNoOfUpdatesOrDeletesOnPrimary;
-    private RetentionLeases retentionLeases;
-
-    public RecoveryTranslogOperationsRequest() {
-    }
+    private final long recoveryId;
+    private final ShardId shardId;
+    private final List<Translog.Operation> operations;
+    private final int totalTranslogOps;
+    private final long maxSeenAutoIdTimestampOnPrimary;
+    private final long maxSeqNoOfUpdatesOrDeletesOnPrimary;
+    private final RetentionLeases retentionLeases;
+    private final long mappingVersion;
```

Comment:
I renamed it to `mappingVersionOnPrimary` and documented it. It's not an upper bound of mapping version since the replica can have a higher mapping version than that parameter when we index those translog operations. Let me know if you still feel `mappingVersionUpperBound` is appropriate.